### PR TITLE
throw on 404

### PR DESCRIPTION
**Pull request by ausfeldt**
# Pull Request Template

## Description

Currently 404 is not treated as failure and returns status code. Its causing confusion with proxy objects + CRUD operations resulting in success.

With stream API's we have the choice of non-exception API. Typed API throwing exceptions on 404 will help with V2 customers migration and also eases get-started scenarios as well.

## Type of change

Please delete options that are not relevant.

- [X ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Closing issues

#513

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/ChangeFeedProcessor/LeaseManagement/DocumentServiceLeaseUpdaterCosmos.cs [hunk 1]
```diff
@@ -81,18 +83,21 @@ public DocumentServiceLeaseUpdaterCosmos(Container container)
...
+                    ItemResponse<DocumentServiceLeaseCore> response = await this.container.ReplaceItemAsync<DocumentServiceLeaseCore>(
+                    id: itemId,
                     item: lease,
                     partitionKey: partitionKey,
                     requestOptions: itemRequestOptions).ConfigureAwait(false);
 
-                if (response.StatusCode == HttpStatusCode.NotFound)
+                    return response.Resource;
+                }
+                catch (CosmosException ex) when (ex.StatusCode == HttpStatusCode.NotFound)
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/ChangeFeedProcessor/LeaseManagement/DocumentServiceLeaseUpdaterCosmos.cs]: Can this be combined with the outer try catch?

File: Microsoft.Azure.Cosmos/src/CosmosClient.cs [hunk 2]
```diff
@@ -429,16 +429,20 @@ public virtual Container GetContainer(string databaseId, string containerId)
...
+                return cosmosDatabaseResponse;                
             }
-
-            cosmosDatabaseResponse = await this.CreateDatabaseAsync(id, throughput, requestOptions, cancellationToken: cancellationToken);
-            if (cosmosDatabaseResponse.StatusCode != HttpStatusCode.Conflict)
+            catch (CosmosException ex) when (ex.StatusCode == HttpStatusCode.NotFound)
             {
-                return cosmosDatabaseResponse;
+                cosmosDatabaseResponse = await this.CreateDatabaseAsync(id, throughput, requestOptions, cancellationToken: cancellationToken);
+                if (cosmosDatabaseResponse.StatusCode != HttpStatusCode.Conflict)
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/CosmosClient.cs]: Conflict is a 409. Won't it throw?

File: Microsoft.Azure.Cosmos/src/Resource/Database/DatabaseCore.cs [hunk 3]
```diff
@@ -232,16 +232,20 @@ internal DatabaseCore()
...
+                return cosmosContainerResponse;                
             }
-
-            cosmosContainerResponse = await this.CreateContainerAsync(containerProperties, throughput, requestOptions, cancellationToken: cancellationToken);
-            if (cosmosContainerResponse.StatusCode != HttpStatusCode.Conflict)
+            catch (CosmosException ex) when (ex.StatusCode == HttpStatusCode.NotFound)
             {
-                return cosmosContainerResponse;
+                cosmosContainerResponse = await this.CreateContainerAsync(containerProperties, throughput, requestOptions, cancellationToken: cancellationToken);
+                if (cosmosContainerResponse.StatusCode != HttpStatusCode.Conflict)
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Resource/Database/DatabaseCore.cs]: Same issue as database. Conflict should will be thrown.

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosItemTests.cs [hunk 4]
```diff
@@ -1462,22 +1474,37 @@ public async Task ContainterReCreateStatelessTest(bool operationBetweenRecreate,
...
-                ResponseMessage response = await iterator.ReadNextAsync();
-                Assert.AreEqual(expected, response.StatusCode, $"ExecuteQueryAsync substatuscode: {response.Headers.SubStatusCode} ");
+                FeedIterator iterator = container.GetItemQueryStreamIterator("select * from r");
+                while (iterator.HasMoreResults)
+                {
+                    ResponseMessage response = await iterator.ReadNextAsync();
+                    Assert.AreEqual(expected, response.StatusCode, $"ExecuteQueryAsync substatuscode: {response.Headers.SubStatusCode} ");
+                }
+            }
+            catch (CosmosException ex) when (ex.StatusCode == HttpStatusCode.NotFound)
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosItemTests.cs]: These are stream apis. We should not be throwing for not found.

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosItemTests.cs [hunk 5]
```diff
@@ -1462,22 +1474,37 @@ public async Task ContainterReCreateStatelessTest(bool operationBetweenRecreate,
...
-                ResponseMessage response = await iterator.ReadNextAsync();
-                Assert.AreEqual(expected, response.StatusCode, $"ExecuteReadFeedAsync substatuscode: {response.Headers.SubStatusCode} ");
+                FeedIterator iterator = container.GetItemQueryStreamIterator();
+                while (iterator.HasMoreResults)
+                {
+                    ResponseMessage response = await iterator.ReadNextAsync();
+                    Assert.AreEqual(expected, response.StatusCode, $"ExecuteReadFeedAsync substatuscode: {response.Headers.SubStatusCode} ");
+                }
             }
+            catch (CosmosException ex) when (ex.StatusCode == HttpStatusCode.NotFound)
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosItemTests.cs]: This is also a stream api and should not throw for not found.

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosNotFoundTests.cs [hunk 6]
```diff
@@ -117,30 +117,30 @@ private async Task ItemOperations(Container container, bool containerNotExist)
             {
                 dynamic randomItem = new { id = "test", pk = "doesnotexist" };
                 Stream create = jsonSerializer.ToStream<dynamic>(randomItem);
-                this.VerifyNotFoundResponse(await container.CreateItemStreamAsync(create, new PartitionKey(randomItem.pk)));
+                await this.VerifyNotFoundErrorAsync(async () => await container.CreateItemStreamAsync(create, new PartitionKey(randomItem.pk)));
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosNotFoundTests.cs]: Please update all the stream apis to not throw on not found.

File: Microsoft.Azure.Cosmos/src/CosmosClient.cs [hunk 7]
```diff
@@ -429,16 +429,20 @@ public virtual Container GetContainer(string databaseId, string containerId)
...
             Database database = this.GetDatabase(id);
-            DatabaseResponse cosmosDatabaseResponse = await database.ReadAsync(cancellationToken: cancellationToken);
-            if (cosmosDatabaseResponse.StatusCode != HttpStatusCode.NotFound)
+            DatabaseResponse cosmosDatabaseResponse;
+
+            try
             {
-                return cosmosDatabaseResponse;
+                cosmosDatabaseResponse = await database.ReadAsync(cancellationToken: cancellationToken);
+                return cosmosDatabaseResponse;
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/CosmosClient.cs]: >                 [](start = 46, length = 16)

nite: Remove extra spaces (pre-exists)

File: Microsoft.Azure.Cosmos/src/CosmosClient.cs (hunk 2 above)
Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos/src/CosmosClient.cs]: Yes it will. 

---
In reply to: [301588875](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/519#discussion_r301588875) [](ancestors = 301588875)

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosItemTests.cs (hunk 5 above)
Comment (ausfeldt) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosItemTests.cs]: That my plan, going to circle back and take care of stream APIs after I get all the emulator tests to pass.

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosItemTests.cs [hunk 8]
```diff
@@ -1693,10 +1720,16 @@ private static async Task TestNonePKForNonExistingContainer(Container container)
             Assert.IsNotNull(response.Headers.ActivityId);
             Assert.IsNotNull(response.ErrorMessage);
 
-            // FOr typed also its not error
-            var typedResponse = await container.ReadItemAsync<string>("id1", Cosmos.PartitionKey.None);
-            Assert.AreEqual(HttpStatusCode.NotFound, typedResponse.StatusCode);
-            Assert.IsNotNull(typedResponse.Headers.ActivityId);
+            // For typed, it will throw
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosItemTests.cs]: Please add 404 to all the typed APIs Exception lists.

File: Microsoft.Azure.Cosmos/src/Handler/RequestMessage.cs [hunk 9]
```diff
@@ -153,11 +153,19 @@ internal async Task AssertPartitioningDetailsAsync(CosmosClient client, Cancella
...
+                ContainerProperties collectionFromCache =
                 await collectionCache.ResolveCollectionAsync(this.ToDocumentServiceRequest(), cancellationToken);
-            if (collectionFromCache.PartitionKey?.Paths?.Count > 0)
+
+                if (collectionFromCache.PartitionKey?.Paths?.Count > 0)
+                {
+                    Debug.Assert(this.AssertPartitioningPropertiesAndHeaders());
+                }
+            }
+            catch (DocumentClientException ex)
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Handler/RequestMessage.cs]: How is this change related to 404?

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosContainerTests.cs [hunk 10]
```diff
@@ -361,8 +361,14 @@ public async Task DeleteNonExistingContainer()
             string containerName = Guid.NewGuid().ToString();
             Container container = this.cosmosDatabase.GetContainer(containerName);
 
-            ContainerResponse containerResponse = await container.DeleteContainerAsync();
-            Assert.AreEqual(HttpStatusCode.NotFound, containerResponse.StatusCode);
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosContainerTests.cs]: Assert.Fail() in try block

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosItemTests.cs (hunk 8 above)
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosItemTests.cs]: >          [](start = 39, length = 9)

nit: extra space

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosNotFoundTests.cs [hunk 11]
```diff
@@ -157,5 +157,17 @@ private void VerifyNotFoundResponse(ResponseMessage response)
             Assert.IsNotNull(response);
             Assert.AreEqual(HttpStatusCode.NotFound, response.StatusCode);
         }
+
+        private async Task VerifyNotFoundErrorAsync(Func<Task> func)
+        {
+            try
+            {
+                await func();
+            }
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosNotFoundTests.cs]: assert.Fail() in try block

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosNotFoundTests.cs [hunk 12]
```diff
@@ -157,5 +157,17 @@ private void VerifyNotFoundResponse(ResponseMessage response)
             Assert.IsNotNull(response);
             Assert.AreEqual(HttpStatusCode.NotFound, response.StatusCode);
         }
+
+        private async Task VerifyNotFoundErrorAsync(Func<Task> func)
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosNotFoundTests.cs]: >VerifyNotFoundErrorAsync [](start = 27, length = 24)

VerifyNotFoundErrorAsync -> VerifyNotFoundExceptionAsync

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/LinqSQLTranslationBaselineTests.cs [hunk 13]
```diff
@@ -39,9 +40,14 @@ public async static Task Initialize(TestContext textContext)
...
-            await cosmosClient.GetDatabase(id: nameof(LinqSQLTranslationBaselineTest)).DeleteAsync();
-            testDb = await cosmosClient.CreateDatabaseAsync(nameof(LinqSQLTranslationBaselineTest));
-
+            try
+            {
+                await cosmosClient.GetDatabase(id: nameof(LinqSQLTranslationBaselineTest)).DeleteAsync();
+            }
+            catch (CosmosException ex) when (ex.StatusCode == HttpStatusCode.NotFound)
+            {
+                testDb = await cosmosClient.CreateDatabaseAsync(nameof(LinqSQLTranslationBaselineTest));
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/LinqSQLTranslationBaselineTests.cs]: outside catch block

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/LinqSQLTranslationBaselineTests.cs [hunk 14]
```diff
@@ -39,9 +40,14 @@ public async static Task Initialize(TestContext textContext)
...
+            }
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/LinqSQLTranslationBaselineTests.cs]: >                        [](start = 14, length = 23)

extra space

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/NameRoutingTests.cs [hunk 15]
```diff
@@ -97,7 +97,16 @@ private async Task SmokeTestForNameAPI(CosmosClient client)
                 string resourceRandom2Id = "randomToDelete2" + suffix;
 
                 // Delete database if exist:
-                Cosmos.Database databaseToDelete = await client.GetDatabase(databaseId).DeleteAsync();
+
+                try
+                {
+                    Cosmos.Database databaseToDelete = await client.GetDatabase(databaseId).DeleteAsync();
+                }
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/NameRoutingTests.cs]: assert.fail()

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/NameRoutingTests.cs [hunk 16]
```diff
@@ -146,8 +155,15 @@ private async Task SmokeTestForNameAPI(CosmosClient client)
 
                     // 
                     await container.ReplaceItemAsync<dynamic>(id: doc3Id, item: new { id = doc3Id, Description = "test" });
-                    doc3 = await container.DeleteItemAsync<Document>(partitionKey: new Cosmos.PartitionKey(resourceRandomId), id: resourceRandomId);
-
+                    try
+                    {
+                        doc3 = await container.DeleteItemAsync<Document>(partitionKey: new Cosmos.PartitionKey(resourceRandomId), id: resourceRandomId);
+                    }
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/NameRoutingTests.cs]: assert.fail()

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/NameRoutingTests.cs [hunk 17]
```diff
@@ -655,6 +671,18 @@ private async Task VerifyGatewayNameIdCacheRefreshPrivateAsync(CosmosClient clie
             }
         }
 
+        private async Task DeleteContainerAsync(Container container)
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/NameRoutingTests.cs]: >DeleteContainerAsync [](start = 27, length = 20)

DeleteContainerAsync -> DeleteContainerIfExistsAsync

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/NameRoutingTests.cs [hunk 18]
```diff
@@ -979,19 +1007,19 @@ private async Task CrazyNameTestPrivateAsync(CosmosClient client, bool useGatewa
 
 
         [TestMethod]
-        public void NameRoutingBadUrlTest()
+        public async Task NameRoutingBadUrlTest()
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/NameRoutingTests.cs]: FI from master will result in conflicts, hence left in original state. Please undo.

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/NameRoutingTests.cs [hunk 19]
```diff
@@ -1051,20 +1075,18 @@ private async Task NameRoutingBadUrlTestPrivateAsync(CosmosClient client, bool b
             catch (CosmosException e)
             {
                 Assert.IsNotNull(e.Message);
-                Assert.IsTrue(e.StatusCode == HttpStatusCode.BadRequest || e.StatusCode == HttpStatusCode.Unauthorized);
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/NameRoutingTests.cs]: Another OR please.

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/NameRoutingTests.cs [hunk 20]
```diff
@@ -1051,20 +1075,18 @@ private async Task NameRoutingBadUrlTestPrivateAsync(CosmosClient client, bool b
...
             {
                 // doing a collection read with Document link
                 ContainerResponse collection1 = await database.GetContainer(UriFactory.CreateDocumentUri(databaseId, collectionId, doc1Id).ToString()).ReadContainerAsync();
-                Assert.IsNull(collection1.Resource);
-                Assert.AreEqual(collection1.StatusCode, HttpStatusCode.NotFound);
             }
             catch (CosmosException e)
             {
                 Assert.IsNotNull(e.Message);
-                Assert.IsTrue(e.StatusCode == HttpStatusCode.BadRequest || e.StatusCode == HttpStatusCode.Unauthorized);
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/NameRoutingTests.cs]: Another OR please not replace

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/HandlerTests.cs [hunk 21]
```diff
@@ -61,8 +61,7 @@ public async Task TestPreProcessingHandler()
 
             HttpStatusCode[] testHttpStatusCodes = new HttpStatusCode[]
                                 {
-                                    HttpStatusCode.OK,
-                                    HttpStatusCode.NotFound
+                                    HttpStatusCode.OK
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/HandlerTests.cs]: >                                    [](start = 54, length = 35)

spces. Please fix editor customization.

File: Microsoft.Azure.Cosmos/src/Handler/RequestMessage.cs (hunk 9 above)
Comment (ausfeldt) [reply Microsoft.Azure.Cosmos/src/Handler/RequestMessage.cs]: The 404 that is thrown is a NotFoundException which inherits DocumentClientException, and the tests are all catching CosmosException.

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/NameRoutingTests.cs (hunk 15 above)
Comment (ausfeldt) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/NameRoutingTests.cs]: not needed, the original code made sure that it was cleaned up, but it can throw now.

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosItemTests.cs (hunk 8 above)
Comment (ausfeldt) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosItemTests.cs]: seems they are already there

File: Microsoft.Azure.Cosmos/src/CosmosClient.cs [hunk 22]
```diff
@@ -429,16 +429,29 @@ public virtual Container GetContainer(string databaseId, string containerId)
...
+
+            if (response.IsSuccessStatusCode)
             {
-                return cosmosDatabaseResponse;
+                return await this.ClientContext.ResponseFactory.CreateDatabaseResponseAsync(database, Task.FromResult(response));
             }
-
-            cosmosDatabaseResponse = await this.CreateDatabaseAsync(id, throughput, requestOptions, cancellationToken: cancellationToken);
-            if (cosmosDatabaseResponse.StatusCode != HttpStatusCode.Conflict)
+            else
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/CosmosClient.cs]: nit: Don't need the else. It causes a lot of indentation which makes it harder to read.

File: Microsoft.Azure.Cosmos/src/CosmosClient.cs [hunk 23]
```diff
@@ -429,16 +429,29 @@ public virtual Container GetContainer(string databaseId, string containerId)
...
+                if (response.StatusCode == HttpStatusCode.NotFound)
+                {
+                    DatabaseProperties databaseProperties = this.PrepareDatabaseProperties(id);
+                    Stream stream = this.ClientContext.PropertiesSerializer.ToStream(databaseProperties);
+                    response = await this.CreateDatabaseStreamInternalAsync(stream, throughput, requestOptions, cancellationToken);
+                    
+                    if (response.StatusCode == HttpStatusCode.Conflict)
+                    {
+                        response = await database.ReadStreamAsync(requestOptions: requestOptions, cancellationToken: cancellationToken);
+                        return await this.ClientContext.ResponseFactory.CreateDatabaseResponseAsync(database, Task.FromResult(response));
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/CosmosClient.cs]: We could have just done a database.ReadAsync instead of needing to do the stream response here.

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosItemTests.cs [hunk 24]
```diff
@@ -116,9 +116,15 @@ public async Task CreateDropItemPartitionKeyNotInTypeTest()
...
-            Assert.AreEqual(HttpStatusCode.NotFound, readResponse.StatusCode);
+            try
+            {
+                readResponse = await this.Container.ReadItemAsync<dynamic>(id: testItem.id, partitionKey: Cosmos.PartitionKey.None);
+                Assert.Fail("Should throw exception.");
+            }
+            catch (CosmosException ex)
+            {
+                Assert.AreEqual(HttpStatusCode.NotFound, ex.StatusCode);
+            }
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosItemTests.cs]: >                                     [](start = 13, length = 36)

extra space. Please fix VS styling

File: Microsoft.Azure.Cosmos/src/Handler/RequestInvokerHandler.cs [hunk 25]
```diff
@@ -72,7 +74,13 @@ public RequestInvokerHandler(CosmosClient client)
             }
 
             await this.client.DocumentClient.EnsureValidClientAsync();
-            await request.AssertPartitioningDetailsAsync(this.client, cancellationToken);
+#if DEBUG
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Handler/RequestInvokerHandler.cs]: Nice

Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Handler/RequestInvokerHandler.cs]: >                  [](start = 9, length = 17)

extra spaces

File: Microsoft.Azure.Cosmos/src/CosmosClient.cs (hunk 22 above)
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/CosmosClient.cs]: Else is not needed.

File: Microsoft.Azure.Cosmos/src/CosmosClient.cs [hunk 26]
```diff
@@ -429,16 +429,29 @@ public virtual Container GetContainer(string databaseId, string containerId)
 
             // Doing a Read before Create will give us better latency for existing databases
             Database database = this.GetDatabase(id);
-            DatabaseResponse cosmosDatabaseResponse = await database.ReadAsync(cancellationToken: cancellationToken);
-            if (cosmosDatabaseResponse.StatusCode != HttpStatusCode.NotFound)
+
+            ResponseMessage response = await database.ReadStreamAsync(requestOptions: requestOptions, cancellationToken: cancellationToken);
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/CosmosClient.cs]: new logix is different from previous one which is returning even its non-success but 404.

**Comment by kirankumarkolli** [comment]
Dupe of #528
